streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorTopologyTest.java

```diff
@@ -367,6 +370,49 @@ public void shouldAddHeaders() {
         assertNextOutputRecord(OUTPUT_TOPIC_1, "key3", "value3", HEADERS, 30L);
     }
 
+    @Test
+    public void testHasPersistentLocalStore() {
```

Comment:
We should split this into multiple smaller tests, each testing one scenario:
 - `statelessTopologyShouldNotHavePersistentLocalStore()`
 - `inMemoryStoreShouldNotResultInPersistentLocalStore()`
 - `persistenLocalStoreShouldBeDetected()`

Or similar test method names that explain what the test is doing / expecting / testing.